Lib/inspect.py

```diff
@@ -2440,11 +2447,12 @@ def __init__(self, name, kind, *, default=_empty, annotation=_empty):
         if kind not in (_POSITIONAL_ONLY, _POSITIONAL_OR_KEYWORD,
                         _VAR_POSITIONAL, _KEYWORD_ONLY, _VAR_KEYWORD):
             raise ValueError("invalid value for 'Parameter.kind' attribute")
-        self._kind = kind
+        self._kind = _ParameterKind(kind)
 
         if default is not _empty:
-            if kind in (_VAR_POSITIONAL, _VAR_KEYWORD):
-                msg = '{} parameters cannot have default values'.format(kind)
+            if self._kind in (_VAR_POSITIONAL, _VAR_KEYWORD):
+                msg = '{!s} parameters cannot have default values'
+                msg = msg.format(_PARAM_NAME_MAPPING[self._kind])
```

Comment:
I'd name the helper `_get_paramkind_descr()`

Reply:
The main benefit of the helper is that its name is shorter and you can write formatting in one line.

In 3.8 I would add an attribute to `_ParameterKind`, and this could be written as `self.kind.descr`. Or maybe override the `__str__` method and write it just as `self.kind`?

Reply:
Thank you for all your kind review. I will update them tonight! Many thanks!